### ISSUE TITLE
v2.3 discussion - please use this for discussion

### ISSUE THREAD

**Issue by JoSchaap** [JoSchaap/GoT_Wasteland_V2.Stratis#85]
**\* Discusscion thread ***

please only discuss here and use the new issue button to report bugs with a full discription and steps to reproduce them

First things first, thanks all for testing, sadly something came up (family thing) so i wasn't able to get any work done yesterday.
What i did find out is that there's more things to fix than i initially thought last night..

The list of issues to fix is below:

[Fixed] - general store broken 

[Fixed] - briefing missing some line breaks

[Fixed] - increase radius of dead player cleanup (items tend to roll away)

[WontChange-isOK] - The 'you have died from starvation/hunger' hints stay on for too long (decrease its duration)

[Fixed] - investigate an issue where players are randomly spawned mid air in limbo when clicking random (findsafepos failing again?)

[Fixed/WorksAsIntended] - investigate iniDB counter in diag_log  'x objects saved' gets stuck and only increases , should be set to 0 after saving

[Fixed] - investigate server cleanup counter in diag_log '9 dead players 34 AI and 30 clutterred items from the dead have been cleaned up' the numbers are no longer re-setting to 0

[Fixed] - revert to the old vehicle.sqf to see if it fixes the newly added logspam

```
        Error in expression <F_LOG_remorque", objNull];
        if (!isNull _towedUnit) then
        {
        detach _towedUnit;
        _t>
        20:30:36   Error position: <_towedUnit) then
        {
        detach _towedUnit;
        _t>
        Error Undefined variable in expression: _towedunit
        mpmissions\__cur_mp.Stratis\server\functions\vehicle.sqf, line 139
```

[Fixed] - A new findsafepos issue has introduced itself?, I thought i got rid of this..

```
        Error in expression <n
        {
        if (!((count (_testPos isFlatEmpty [_objDist, 0, _maxGradient, _objDist max >
        Error position: <_objDist, 0, _maxGradient, _objDist max >
        Error Undefined variable in expression: _objdist
        File A3\functions_f\misc\fn_findSafePos.sqf, line 100
        Error in expression <};
        };
        _pos = [_markerPos, 2, 45, ( if (_type == 1) then { 2 } else { 5 } ), 0, >
        Error position: <_type == 1) then { 2 } else { 5 } ), 0, >
        Error Undefined variable in expression: _type
        File mpmissions\__cur_mp.Stratis\server\spawning\vehicleCreation.sqf, line 30
        Error in expression <n
        {
        if (!((count (_testPos isFlatEmpty [_objDist, 0, _maxGradient, _objDist max >
        Error position: <_objDist, 0, _maxGradient, _objDist max >
        Error Undefined variable in expression: _objdist
        File A3\functions_f\misc\fn_findSafePos.sqf, line 100
```

[Fixed] - various new undefined variable spam in rpt (not sure why bis changed this..)

```
        Cannot create non-ai vehicle ,
        Error in expression <50);
        _car setDamage (random 0.50);
        if (_type in [0,1]) then
        {
        _car setHit ["whe>
        Error position: <_type in [0,1]) then
        {
        _car setHit ["whe>
        Error Undefined variable in expression: _type
        File mpmissions\__cur_mp.Stratis\server\spawning\vehicleCreation.sqf, line 44

        Error in expression <ion Failed: %1",_missionType];
        } else {
        _vehicle setVehicleLock "UNLOCKED";
        _veh>
        Error position: <_vehicle setVehicleLock "UNLOCKED";
        _veh>
        Error Undefined variable in expression: _vehicle
        File mpmissions\__cur_mp.Stratis\server\missions\mainMissions\mission_Convoy.sqf, line 137
```

[Fixed] - Main mission: coastal patrol, remove the SDV it keeps getting beached by the AI
*\* investigate why AI tends to beach the boat when engaging players

[Fixed] - extend the time between loot re-spawns and increase precision

[Fixed] - missions no longer delete blown up vehicles..

```
        Error in expression <%1",_missionType];
        } else {
        if ((damage _vehicle) == 1) then {
        deleteVehicle _ve>
        Error position: <_vehicle) == 1) then {
        deleteVehicle _ve>
        Error Undefined variable in expression: _vehicle
        File mpmissions\__cur_mp.Stratis\server\missions\mainMissions\mission_Coastal_Convoy.sqf, line 180
```

[Fixed] - Clientsided errorspam when in a group and looking at a group member (screen shots below)
[Fixed] - clientsided errorspam when in bluefor/opfor and looking at a group member
(screen shots below)

[Fixed] - check if bis_fnc_rsclayer or fn_dynamictext has been changed (sometimes causes an error)

[Fixed] - client sided error message when picking up money while to far away from it (screenshot below)

[Should-be-Fixed(TEST!)] - newplayericons ocasionally still drops a clientsided blackbox error about an undefined hud_icon (see jackiechan image below)

[Fixed] - old undefined variable error is back when you stop spectating a player (probably overwritten my own fixes)

![2013-07-26_00010](https://f.cloud.github.com/assets/2301588/866722/e090a66e-f6a8-11e2-81cf-844f9b4bc3ce.jpg)
![2013-07-26_00011](https://f.cloud.github.com/assets/2301588/866723/e094604c-f6a8-11e2-822b-6e95f4d1768f.jpg)
![2013-07-25_00004](https://f.cloud.github.com/assets/2301588/866724/e09294d8-f6a8-11e2-8dcd-97273ca5c4ee.jpg)
![2013-07-26_00008](https://f.cloud.github.com/assets/2301588/866725/e09353dc-f6a8-11e2-8cd9-4991df41a203.jpg)
![2013-07-26_00012](https://f.cloud.github.com/assets/2301588/866726/e091ced6-f6a8-11e2-8587-ee8cfb202aeb.jpg)
![2013-07-26_00009](https://f.cloud.github.com/assets/2301588/866727/e096ac30-f6a8-11e2-91e7-c26d0b96703f.jpg)
![2013-07-26_00013](https://f.cloud.github.com/assets/2301588/866728/e0a2e07c-f6a8-11e2-9a38-7f0b25bd94ba.jpg)
![2013-07-26_00018](https://f.cloud.github.com/assets/2301588/866730/e0a48576-f6a8-11e2-9c1b-aadb8063129c.jpg)
![2013-07-26_00014](https://f.cloud.github.com/assets/2301588/866729/e0a73794-f6a8-11e2-8f1f-4f922b395fdf.jpg)
![2013-07-26_00019](https://f.cloud.github.com/assets/2301588/866731/e0a53f0c-f6a8-11e2-8910-d11544f663ad.jpg)
![2013-07-26_00020](https://f.cloud.github.com/assets/2301588/866732/e0a418c0-f6a8-11e2-9023-7c6c8ca1b8dd.jpg)
![2013-07-27_00004](https://f.cloud.github.com/assets/2301588/866733/e0aa24ea-f6a8-11e2-9f20-4b68d3445d10.jpg)
![jackie-chan-meme](https://f.cloud.github.com/assets/2301588/866735/ec5c3a76-f6a8-11e2-9d51-2db78038e835.jpg)

**Comment by SCETheFuzz** [comment]
Let me know when you need test guys again. I'm off to bed for about 5 hours of sleep

**Comment by s3kShUn61** [comment]
Oh one little thing would be Katiba ammo added to the gunstore. It currently is not available.

**Comment by s3kShUn61** [comment]
So any eta on release? Im losing a lot of clients to these error messages and a release would be greatly appreciated even if it has a few left. Any chance 2.3testing could be merged with the main for a .zip dl? Then fixes over the next week or so?

**Comment by JoSchaap** [comment]
I'm not giving an ETA before i find out what causes the wierd errorboxes without any error being present. ive broken over 3 deadlines allready :)

tonight ill be continuing the hunt for these issues :)

**Comment by MercyfulFate** [comment]
does this mean that the issues you had with my inventory system might not be from the system itself?

**Comment by s3kShUn61** [comment]
What is wrong with the errorboxes? I dont know what you saw about it but from what Ive been reading the undefined variables are from a Nil value which causes the variable to become undefined. You probably knew this already but Ive been looking into fixing the 2.3 I currently have since its just a constant stream of these errors.

**Comment by JoSchaap** [comment]
the undefined variable errors are being triggered by defined and set variables, thats why i included the jackiechan image at the bottom of the image..

```
    private ["_variable"];
    _variable = 1;
    if (_variable == 1) then { diag_log "it works!"; };
```

run this ingame and it gives 'Undefined variable "_variable" on line 2'

... i guess my programming logic is just wrong but imho theres nothing undefined here..
then again adding isnill checks removes the black error box, but also skips the code...

**Comment by s3kShUn61** [comment]
Well from my work in Java what I've seen shows that the compiler is seeing a variable, let's say _x, and that it is being set to something that has to be accessed, such as a function call to define the variable. The compiler in c++ (I believe but I am not sure) is seeing variable _x=_someFunctionCallToSetX_ and saying there is a chance _x can be null if the method call fails or returns null. A fix would be to set _x to something definite, like 0 or whatever type _x is, before the actual _x=_someFunctionCallToSetX_. This will always ensure that _x is something to the compiler even if _someFunctionCallToSetX_ returns null or fails.

**Comment by AgentRev** [comment]
From the tests I've done, it has nothing to do with `private` assignments, undefined variable means that the variable is `nil`.

I've only had a few errors in my mission, most of them had to do with mistyped variables, badly-structured code, or `getVariable` calls. I didn't get any of the specific errors from the screenshots above.

I can't possibly understand how the `_variable` test could possibly say undefined. Did you actually try this very code in the game?

**Comment by s3kShUn61** [comment]
This is what ive found from a forum post:
"I think lot of the problems with undefined variables in these scripts are due to the use of the nil command in the scripts. For example in the scripts: _milGroup = if (count _this > 9) then { _this select 9; } else {nil;};

From the wiki: Nil value. This value can be used to undefine existing variables.

This was fine in previous builds, but with the new behaviour of Arma to throw an error when a variable is undefined means that the whole way the scripts work generates lots of undefined variable errors as the author builds the core functionality on testing to see if a variable exists or is undefined. Don't get me wrong, whoever wrote it is very clever, but unfortunately, his method for the scripts has become verboten."

**Comment by s3kShUn61** [comment]
And no I have not had a chance to test it, it was only an idea. Ive been very busy the past few weeks and I have had very little time to do anything.

**Comment by JoSchaap** [comment]
@AgentRev  im typing on my cellphone (github app) so no i cant test, im on a train atm :+1: 
but i guess i just need to look at it from a big screen, coding on my tablet isnt going that well either. :)

will get back on this when im at home :)

**Comment by AgentRev** [comment]
Certain errors can be legitly corrected by using `isNil` (like `getVariable` errors), but it's not a good idea to do that everywhere, as it may hide more serious underlying issues.

About the `dynamicText` errors above, I'm a bit clueless, since those are core game files, so it's probably BIS that didn't do its job.

About the `pickupMoney` error, the answer is simple. The "Pickup money" action is shown when a moneybag is within 5m, but the actual pickup script only detects moneybags that are within 3m. Silly 404Games. The fix for this one is, at line 18 in `client\actions\pickupMoney.sqf`, set the `3` to something higher than 5 for margin, like `6`.

**Comment by JoSchaap** [comment]
good lets see...

46f2f2ff75f8b06b610c8bb6738d123d619d8513
f768749133908c2ca926b39d513ccd68493246a3
7bebe8e1a02a2e5f21811440c9379236115f3765
106621cdf22d77aaca9d79581d6d1edeafa733bc
05b0662a6b12e356b13fb2d2131eacae83a7ffb1
92cdb1e474c26d4d1f6e8869bf094c87801728e8
2b856f8bf5197d6268fd7a46e55762bf74a163e5
95d46fadc2c86cb3d93384d9feaabf47a3405617

@agentrev dynamictext was due to grouptags not having _refresh defined...

i still have no clue about the random mid-air limbo respawns though

**Comment by MercyfulFate** [comment]
sounds like something going wrong in the arma engine, rather than the scripting. I cant seem to reproduce any errors (other than the mid air-respawns) on either 2.3 code, or my inventory code.

**Comment by AgentRev** [comment]
Most of these errors are related to interactions with other live players, so if you are playing alone, you most likely won't be able to reproduce them.

<img src="http://www.i2clipart.com/cliparts/5/4/0/0/clipart-forever-alone-5400.png"/>

**Comment by JoSchaap** [comment]
@s3kShUn61 thanks for joining the game so i could test the tags changes :P

**Comment by s3kShUn61** [comment]
@JoSchaap Np man, anything to help get this build ready :D

**Comment by SCETheFuzz** [comment]
You guys still in server?

**Comment by JoSchaap** [comment]
no but the help was usefull new build comming up soon :)

**Comment by JoSchaap** [comment]
ok so the only issue left is to fix the hud_icon error.

i thought it was just me having this. but im currently watching a livestream of someone playing on the TPG server and he's also getting the error..

![errorpopup](https://f.cloud.github.com/assets/2301588/875677/3d6fb314-f8a9-11e2-9416-fce506c3e770.jpg)

once this one has a resolution (but ive tried about anything allready)
its not far from making a release :)

**Comment by s3kShUn61** [comment]
Got one for ya, my admins showed me how you can access the gunstore menu, then continue to walk while the menu is open allowing a player to leave the gunstore and buy items. Just set a proximity like picking up money?

**Comment by SCETheFuzz** [comment]
Can we test out the player team unlock ?

**Comment by JoSchaap** [comment]
Team unlock should work.

As for the stores. Ill see if i can lock the player so movement is out of the question :)

**Comment by s3kShUn61** [comment]
Can you also lock the gunstore AI? I know movement is disabled but players can push them away using vehicles. We were able to push him to our base XD

**Comment by JoSchaap** [comment]
hehehe, i'm not sure, have to look into that :)

**Comment by AgentRev** [comment]
I've looked at a lot of stuff regarding the error in the last screenshot above. If I play alone and I start creating friendly units using the Proving Grounds console, the error always appears after I spawn the 46th unit. I've increased the player icons in `clients\system\newicons.hpp` to 100, but it still shows.

My theory would be that something is being accessed too often when refreshing the icons, and it makes the script unstable. The best bet would to be simply add `isNil`'s before every reference to `FZF_IC_Hud_Disp` and `HUD_ICON` in `client\functions\newPlayerIcons.sqf`, and probably refactor the script too because it's especially inefficient, not to say a pile of junk.

**Comment by JoSchaap** [comment]
ok new testbuild up on the servers

password
        borrelnoot

@AgentRev, hud_icon seems to be an issue in all missions that use it :/ even seen it pop up on Sama's version while watching streams yesterday

**Comment by AgentRev** [comment]
I will rewrite `newPlayerIcons.sqf` and see how that goes.

**Comment by AgentRev** [comment]
An idea to prevent gunstore NPCs from moving would to "glue" them to the building they are in with `attachTo`. Will write something.

**Comment by s3kShUn61** [comment]
Will they stay after the building is destroyed. What happens as far as I know is the airbase gunstore building is no-clipped after being destroyed. I dont know if that will affect attachTo.

**Comment by AgentRev** [comment]
Hmm you're right... I was thinking about making gunstore buildings invulnerable, thought.

**Comment by AgentRev** [comment]
Something like this added to the NPC's init field might do it:

```
_gunStore = nearestObject [this, 'Building']; _gunStore addEventHandler ['HandleDamage', {false}]; this attachTo [_gunStore, _gunStore worldToModel getPos this];
```

The only things I'm not sure of are if the NPC will keep facing the same direction, and if having a private variable (`_gunStore`) in an init field is possible, will have to test.

**Comment by s3kShUn61** [comment]
Making the buildings invulnerable would be a great idea. I've quite a few asshats who like to say they are attacking a player with RPGs and will blow up all the gunstore buildings. That would be an easy solution.

**Comment by JoSchaap** [comment]
or just ad some setpos commands to the cleanup that reverts the gunstores to their original location/direction? :)

or add them to a respawn like vehicles :P

**Comment by AgentRev** [comment]
Found something much simpler on the BIS forums:

> Simply put `this enableSimulation false` in the plane's init field. That sucker won't budge. Ram into it, shoot it, nuke it... it's not gonna bat an eyelash at you.

Will have to test.

**Comment by JoSchaap** [comment]
That should fix it indeed :)

**Comment by JoSchaap** [comment]
seems to work sofar the ai didnt budge :)

**Comment by SCETheFuzz** [comment]
http://steamcommunity.com/sharedfiles/filedetails/?id=165222385 

from selling my gear at the gun store.

**Comment by JoSchaap** [comment]
Nice find :)

**Comment by AgentRev** [comment]
Yeah that's my fault, I mixed up the S and Z. Just replace all `_magasines` to `_magazines`. Maybe I should actually commit corrections I make after writing them, lol.

**Comment by JoSchaap** [comment]
i allready spotted that :) but thanks for the reply :)

**Comment by SCETheFuzz** [comment]
Need another round of testing?

**Comment by JoSchaap** [comment]
its on the server if youre bored :)

**Comment by JoSchaap** [comment]
updated opening message, all should be fixed, just need some ppl to test if any new errors occurr (at work, cant test)

both servers are running the build now

password:

```
       borrelnoot
```

**Comment by SCETheFuzz** [comment]
Will jump on right now maybe let us test drive it on our servers in say 9 hours after I wake up?

**Comment by SCETheFuzz** [comment]
http://cloud-2.steampowered.com/ugc/920143977343749850/5BBA6C6C214106504B6884D1361CF271693FBECD/
http://cloud.steampowered.com/ugc/920143977343747359/C4CB929CE7A430A99E52600A8C883D2D5C519820/

Also Blue team and Red team have no team markers 

Going to make you a base to look at when you get home :)

**Comment by JoSchaap** [comment]
The hud error is something i forgot to remove. Will fix that soon. But the error you get in the lobby is just wierd. :)

**Comment by JoSchaap** [comment]
ok the hud error has been fixed in the source: 52345e25a7f429a69c8ebe80addfeecd79bc2824

@SCETheFuzz can you provide me the logs about the error in the second screenshot,

the clientsided RPT logfiles are found by browsing to

```
      %APPDATA%\..\Local\Arma 3\
```

using windows explorer (just paste it into the location bar)

you can also pastebin the entire logfile (just from the past hour) and post a link here :)

**Comment by AgentRev** [comment]
The lobby error is BIS' fault.

**Comment by AgentRev** [comment]
You forgot to change all the the variable names in newPlayerIcons.sqf:
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commit/7dd9724b271eec5ff24275eef1f98c29043d883f

**Comment by SCETheFuzz** [comment]
Still want the lobby error?

**Comment by JoSchaap** [comment]
@SCETheFuzz yes please because the game didnt update today so it is being caused by the mission im affraid :)

@AgentRev i didnt notice the underscores, thanks!

**Comment by SCETheFuzz** [comment]
http://pastebin.com/peRwtXyV

Have at it, also @AgentRev part of the error has the anti hacks in it. You should know that after mission start I was not able to reproduce the error.

**Comment by JoSchaap** [comment]
the logspam was due to me forgetting to rename some vars however the reason for the lobby error i cant seem to find.
ive updated the servers im curious if it still happens :)

**Comment by s3kShUn61** [comment]
Im in now and I havent seen anything odd.

**Comment by AgentRev** [comment]
Can you make a <a href="https://gist.github.com/">Gist</a> instead of using Pastebin? That site is blocked from work...

**Comment by TBGGrapidyGrape** [comment]
Jo I will be gone most day .. Will hop in and test later this afternoon .

**Comment by JoSchaap** [comment]
@AgentRev you vould use a proxy like http://www.hidemyass.com ;)
ahwell here it is: https://gist.github.com/JoSchaap/6122562

@s3kShUn61 good to hear, guess ill see it myself later tonight

@TBG not a problem life comes first ;)

**Comment by AgentRev** [comment]
My work firewall is very aggressive, everything related to proxies or VPNs is blocked. Even some legit corporate websites are blocked, being marked as "SaaS and B2B", whatever that means. I can't even connect to my home computer via TeamViewer.

I've tried searching what `disableSerialization` actually does, but this is the only thing I found:

> Nobody knows what it is for, which doesn't help. Might be something to do with MP savegames.

It's most likely because running scripts are serialized when saving the game, but that UI types can't be serialized.

**Comment by JoSchaap** [comment]
shouldnt you be eating cake? :birthday:

**Comment by s3kShUn61** [comment]
hidemyass.com?? ew no. Use EarthVPN.com. They dont report activity to authorities. HMA is not a true VPN since your data isnt private... XD

**Comment by AgentRev** [comment]
All VPN ports are blocked on outbound. Anyway, I'm not gonna pay for a proxy or VPN just for the workplace when I can wait a couple hours and do it home. Lol.

**Comment by JoSchaap** [comment]
HMA is just something from my highschool dayz 14 years ago ;)

@AgentRev i thought it allready had that, will add it now :)

**Comment by s3kShUn61** [comment]
I know I've asked alot, but possible release tonight? Everything looks to be good now! Havent seen any errors.

**Comment by AgentRev** [comment]
Improved money pick up, cancels as soon as you get too far, as opposed to the end of the script, and doesn't switch bag mid-way : 
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commit/1e74f7795e895d692b354a9db6ab4e4d2acb39a5
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commit/6b21a3bbdbd4d97583024f252c97f9506d4a593e

**Comment by JoSchaap** [comment]
@AgentRev  nice one :)

it still spams this on the clientsided RPT

```
   Warning Message: Variable '_fzf_ic_hud_disp' does not support serialization. Call 'disableSerialization' in the current script (maybe '') if you need to use it.
   Warning Message: Variable '_hud_icon' does not support serialization. Call 'disableSerialization' in the current script (maybe '') if you need to use it.
   Warning Message: Variable '_fzf_ic_hud_disp' does not support serialization. Call 'disableSerialization' in the current script (maybe '') if you need to use it.
   Warning Message: Variable '_hud_icon' does not support serialization. Call 'disableSerialization' in the current script (maybe '') if you need to use it.
```

seems to just be a warning not sure if it impacts client performance

**Comment by AgentRev** [comment]
I think I know the problem, try moving `disableSerialization` just before <a href="https://github.com/JoSchaap/GoT_Wasteland_V2.Stratis/blob/%23TEMP_TESTING_V23/client/functions/newPlayerIcons.sqf#L168">line 168</a>.

**Comment by JoSchaap** [comment]
seems to work :) thanks

**Comment by JoSchaap** [comment]
just need to check if player icons are working now, anyone available to join me? :)

**Comment by s3kShUn61** [comment]
yup be on in a sec

**Comment by AgentRev** [comment]
Some other changes and bugfixes that Jo might find useful:
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commits/master

**Comment by SCETheFuzz** [comment]
http://cloud-3.steampowered.com/ugc/921270022621753399/D03CAACB612B4E553D4A5F82B33FE10E58C1D24A/

**Comment by JoSchaap** [comment]
nice find, didnt click that button during tests :)

did it find hacked vehicles before you clicked it?
because i think it's just an !isnil thing

**Comment by SCETheFuzz** [comment]
http://steamcommunity.com/sharedfiles/filedetails/?id=165695759
http://steamcommunity.com/sharedfiles/filedetails/?id=165694992

As you can see I'm naked to other players ! :(  This is from buying gear from the gun store from sniper to dive gear. I sold each part off by its self then added bought the new stuff.

**Comment by JoSchaap** [comment]
You can only wear the ghilly / diver suit from your own team! Thats why the teamname is behind it.
Or wait till arma3 fixes it.. :)

**Comment by SCETheFuzz** [comment]
http://cloud-4.steampowered.com/ugc/612773448167533908/1FB3ABA97A22F68D867CA2487ECD56788A80E103/

Wont get this error all the time but I do get it from time to time with the off road. 

Also may want to look into the clean up script getting the AI groups, finding a lot of objectives getting cleared with no one in the server with me and no dead body's around.

**Comment by JoSchaap** [comment]
cleanup only deleted dead bodies and empty groups

the error on the offroad looks like a arma3 bug, i dont use sethit in the mission code

**Comment by s3kShUn61** [comment]
Now whats happening with basesaving? I enabled the mod with -mod=@inidb paramater and set basesaving to 1 in your config file but it did not work. When I locked base parts then did a restart they were all reset. Is there more that must be done to enable the mod?

**Comment by AgentRev** [comment]
@JoSchaap for ghillies and wetsuits:
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/blob/master/client/systems/gunStore/buyGuns.sqf#L139-174

**Comment by JoSchaap** [comment]
@s3kShUn61 do you see it saving/loading in the servers RPT logs?
has inidb made a \db\anyObjects.ini yet?

it looks arround the map for locked objects every 5 minutes and saves them to inidb

@Rev
i know you did that but i cant jus go on and import the entire mission :package: 
i think the teamname behind the suits should suffice :)

**Comment by AgentRev** [comment]
> i think the teamname behind the suits should suffice :)

Well, s3kShUn61 kinda proved you it isn't sufficient, didn't he?

**Comment by s3kShUn61** [comment]
Okay I didnt realize it was a 5 min sweep to build the saved base parts file. I was just locking 2 or 3 then immediatly restarting XD so it works great now!

**Comment by contactgus** [comment]
Working fine, thanks. Base-saving servers are showing up as yellow with a question mark in the server browser which is due to the inidb addon. I reckon that might be affecting the popularity as there doesn't seem to be many takers today. The lack of vehicles is not going down too well but we'll stick with it and see if the improved vehicle clean-up makes up for it

**Comment by AgentRev** [comment]
Yeah the yellow icon kinda sucks, BIS should implement something that allows addons to tell the game they are server-only, so that it shows a normal icon and doesn't warn users they don't have it installed on joining.

**Comment by s3kShUn61** [comment]
@contactgus I increased the number of vehicles in my server, to do this just navigate to x.stratis\server\functions\vehicleSpawning and in line 13, which is 

```
for "_i1" from 0 to 134 step 2 do
```

remove the "step 2" so it reads

```
for "_i1" from 0 to 134 do
```

**Comment by contactgus** [comment]
nice one, thanks.

**Comment by s3kShUn61** [comment]
Also, the gunstore glitches are still there. The ones I'm talking about are rapid spamming purchases when the inventory is filled with 6 or so lowcost items. This also starts to give the players a ton of money since it returns money when the inventory is full however they are not being charged for the items.

**Comment by JoSchaap** [comment]
wait.. so they arent getting the 'dont spam the purchase button' message? because it wont do anything with the money before they gone past that :/

**Comment by s3kShUn61** [comment]
Nope... I just tried it and got 45 Stanag mags and it didnt cost me anything

**Comment by AgentRev** [comment]
Jo, I think you made your active purchase variable private instead of global, therefore the value doesn't get passed around between purchase script instances.

**Comment by Krunch** [comment]
@AgentRev You can have server-side mods by adding them to the main addons folder. That is what I do to have CBA server-side only. That is how I implemented my message rotation to work for any mission without being scripted inside each mission... and GREEN icon!

**Comment by s3kShUn61** [comment]
@JoSchaap So move the entire @inidb mod folder to addons then set -mod=Addons\@inidb ?

**Comment by JoSchaap** [comment]
@s3kShUn61 i have always run mods as they should, by using the -mod parameter, i cant surely answer if it would cause any issues or not

**Comment by Krunch** [comment]
@s3kShUn61 move the content of @inidb/addons into the main addons folder, no need to use -mod parameter. It works well with CBA, did not try with iniDB.

**Comment by s3kShUn61** [comment]
Well i've been trying a few things using your recommendation @Krunch however iniDB nolonger seems to save locked base parts. The anyparts file has not been updated in 20 mins and the parts are not there after a restart 20 mins after locking them.

**Comment by SCETheFuzz** [comment]
We noticed a large hit in performance on our server in this last update only getting about 4 hours of out it at 40 people. Where as before we were running 10 hours with out the server given players red chain.

**Comment by JoSchaap** [comment]
thats wierd because in all previous versions the cleanup did not function at all.
if you scan the logs at the period of the red-chains, are you seing the maintenance routing cleaning up bodies and clutterred items? or are they stuck on 0 ?

the only recent change was converting the cleanup from FSM to SQF, but that should not make a difference at all.

**Comment by SCETheFuzz** [comment]
They are stuck on 0 and its not cleaning up the items like jerry cans, money or repair kits even with the area on the clean up pushed out to 100. I ended up dropping down the # of total cars in the server to improve the performance I cut it so the starting amount of cars were 80 ish and after 6 hours we are down to 30  however we are now keeping 30fps for all 40 players for almost 7-8 hours of continuous play. (before---->   After about 6 hours we went from the starting 150 to 180 cars with more then 60 of them having damage of 1 and not being removed over time.

**Comment by AgentRev** [comment]
Maybe try to remove the `local _x` at <a href="https://github.com/JoSchaap/GoT_Wasteland_V2.Stratis/blob/master/server/WastelandServClean.sqf#L24">line 24</a>:

```
if (time - (_x getVariable ["processedDeath", time]) > _deathTime) then
```

The bodies probably aren't local to the server, it could be why it doesn't work.

**Comment by SCETheFuzz** [comment]
The bodies are gone its just the other stuff getting left behind.

**Comment by AgentRev** [comment]
The bodies are already disappearing by default without the clean-up script. If a body disappears before the script had the time to clean its area, it won't be able to, since the body is no more in `allDead`. It could be another reason as to why it doesn't work.

**Comment by AgentRev** [comment]
Ultimately, it might be a more effective idea to attach individual clean-up scripts to dropped items, something similar to what I made for mission vehicles: https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/blob/master/server/functions/cleanVehicleWreck.sqf

**Comment by JoSchaap** [comment]
thats also an idea, and could even be set to the items when they get dropped :)
or just store the location of the body's in an array upon death and cleanup the areas of those locations, then clear the array

**Comment by AgentRev** [comment]
I'm not sure if it's a good idea to add cleanup to items dropped manually. Sometimes when I go attack a base, I drop the bulk of my inventory and money in a hidden location, and I go pick them back up an hour later. Maybe apply it only to Wasteland items, but not money.

For body locations, you'd need to store the position, the time of death, and the player UID belonging to the body. What is also needed is an `originalOwner` variable added to dropped items containing the player UID, so that if someone decides to drop something near a body, it won't disappear if the body is cleaned up like a minute later.

**Comment by JoSchaap** [comment]
@AgentRev i meanth when dropped from the OnKilled script ;) just put a timestamp in a variable or something and let the cleanup chech it just like the death-time on corpses instead of a radius arround the corpse :)

**Comment by AgentRev** [comment]
The radius would be better, since iterating over `allDead` or a manual array of corpses is much more efficient than iterating over something like `allMissionObjects "Thing"`, which consumes a lot more server resources simply to be generated.

**Comment by JoSchaap** [comment]
But if a player respawned or died when near the spot where you just dropped your money and food. You'll lose that too..

**Comment by AgentRev** [comment]
> But if a player respawned or died when near the spot where you just dropped your money and food. You'll lose that too..

<b>↓↓↓</b>

> What is also needed is an originalOwner variable added to dropped items containing the player UID, so that if someone decides to drop something near a body, it won't disappear if the body is cleaned up like a minute later.

**Comment by SCETheFuzz** [comment]
http://i.imgur.com/6LEWLLa.jpg

I think this is on the known bug list where you cant accept group invites even after leaving the server.

**Comment by JoSchaap** [comment]
@SCETheFuzz it wasnt but ive fixed it in v2.4 so thanks for reporting

**Comment by AgentRev** [comment]
Here are some things you might find useful :

https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commit/130920832739776b926a9acd6bb2499a36945ac9
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commit/62849def912dc0c04a925bb11526aaa72b9269a0
https://github.com/AgentRev/Arma3-404Wasteland-Stratis-TPG/commit/83227d6a14039e3c3dd564c0f84b29764e1f1ba3

**Comment by JoSchaap** [comment]
i allready have uniform and vest selling in the store :) but the other 2 i will look into :)

**Comment by AgentRev** [comment]
I know that, but I made so it also gives you the correct amount of money for every item in your uniform, and for magazines, prorate the amount by the ammo quantity left in each, along with a confirmation dialog that shows profits and the list of items that will be sold.

For Sell Current Weapon, it gives you the correct amount for each of your attachments, and prorated magazines too. If you remove the magazine from the weapon, it will not sell the magazines in your inventory.

I worked all weekend on this, you should try it.

**Comment by AgentRev** [comment]
The only thing I forgot for sell weapon is to take into account if something is loaded in the grenade launcher when present.

**Comment by SCETheFuzz** [comment]
Maybe make the mission able to support up to 70-80 players?

**Comment by AgentRev** [comment]
Mine supports 84 players, I was gonna merge that when we'd get to Altis.

**Comment by SCETheFuzz** [comment]
Jo when you have time I would like to talk to you on teamspeak or something about the clean up script more. Server has been up for 13 hours with about 40 ish players and is doing a lot of guns stacked 3-4 deep on spawn spots, and body's staying around for about an hour ish.

**Comment by JoSchaap** [comment]
during the conversion to altis we will rewrite the mechanics of most of the serversided code and this is allready on our list, if this issue still occurs after release of the new version we will be looking into this.

the GoT repositories are going to be archived after releasint the new version and a stratis edition of Wasteland by Team-Wasteland will then also be made :)

**Comment by SCETheFuzz** [comment]
Yeah I have been keeping tabs on the altis project :D. Could you link me to a spot so I can see kinda whats planned to be in it? ?

Never mind found it :D

**Comment by JoSchaap** [comment]
any public reveal will be done on the different forums (ours Bis, 404 etc)

http://forums.a3wasteland.com/index.php?topic=3.msg8#msg8
